sw/device/sca/aes_serial.c

```diff
@@ -607,24 +673,49 @@ static void aes_serial_set_default_values(const uint8_t *data,
   SS_CHECK(data_len == sizeof(uint32_t));
   uint32_t command = 0;
   command = read_32(data);
-  static const uint8_t kPlaintextFixedStart[kAesTextLength] = {
-      0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa,
-      0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa};
-  static const uint8_t kKeyFixedStart[kAesTextLength] = {
+  // Starting constants for fixed-vs-random key, DTR Section 5.3
+  static const uint8_t kPlaintextFixedStartFvsrKey[kAesTextLength] = {
+      0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA,
+      0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA};
+  static const uint8_t kKeyFixedStartFvsrKey[kAesTextLength] = {
       0x81, 0x1E, 0x37, 0x31, 0xB0, 0x12, 0x0A, 0x78,
       0x42, 0x78, 0x1E, 0x22, 0xB2, 0x5C, 0xDD, 0xF9};
-  static const uint8_t kPlaintextRandomStart[kAesTextLength] = {
+  static const uint8_t kPlaintextRandomStartFvsrKey[kAesTextLength] = {
       0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc,
       0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc};
-  static const uint8_t kKeyRandomStart[kAesTextLength] = {
+  static const uint8_t kKeyRandomStartFvsrKey[kAesTextLength] = {
       0x53, 0x53, 0x53, 0x53, 0x53, 0x53, 0x53, 0x53,
       0x53, 0x53, 0x53, 0x53, 0x53, 0x53, 0x53, 0x53};
+  // Starting constants for fixed-vs-random data, DTR Section 5.1
+  static const uint8_t kPlaintextFixedStartFvsrData[kAesTextLength] = {
+      0xDA, 0x39, 0xA3, 0xEE, 0x5E, 0x6B, 0x4B, 0x0D,
+      0x32, 0x55, 0xBF, 0xEF, 0x95, 0x60, 0x18, 0x90};
+  static const uint8_t kPlaintextRandomStartFvsrData[kAesTextLength] = {
+      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+  static const uint8_t kKeyStartFvsrData[kAesTextLength] = {
+      0x01, 0x23, 0x45, 0x67, 0x89, 0xAB, 0xCD, 0xEF,
+      0x12, 0x34, 0x56, 0x78, 0x9A, 0xBC, 0xDE, 0xF0};
+
+  // Initial state of the prng
+  static const uint32_t kPrngInitialState = 0x99999999;
```

Comment:
It would be great if this initial state could be provided via simpleserial. Because right now it's fixed at compile time. As this PR already alters the command, you could easily also add a second uint32 value with the LFSR seed. Do you think this would be feasible?

Reply:
I've now added a separate command `aes_serial_seed_lfsr_order` to re-seed the second LFSR.
This way, it's easier to handle from the ot-sca side. The user just needs to run this command after `aes_serial_set_default_values` and before `aes_serial_fvsr_data_batch_encrypt`.

I've also kept the default value of `0x99999999` in case that the user doesn't want to use this feature.